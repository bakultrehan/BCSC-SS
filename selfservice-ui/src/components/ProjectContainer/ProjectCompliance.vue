/** * ProjectCompliance of app */

<template>
  <v-container>
    <v-row class="ma-5">
      <v-col cols="12">
        <v-card flat>
          <h2 class="text-left tab-headline page-title-h2">
            {{ $t('projectCompliance.pageTitle') }}
          </h2>
          <p class="text-left pt-5">
            After successful experimentation, if your BPS team wants to pursue
            an integration with real citizen BC Services Cards, a BPS team
            member <strong>triggers</strong> a promotion of the project’s
            <em>Development Access Request</em> into a
            <em>Production Access Request</em>.
          </p>
          <p class="text-left">
            The Provincial IDIM Program is mandated to provide identity and
            authentication services using the BC Services Card to support
            citizen transactions with government services.
          </p>
          <p class="text-left">
            Before any Production Access occurs, Open ID Connect (OIDC)
            technical details are provided by the BPS team and a series of
            checks and balances are needed to provide the IDIM Program with the
            confidence required to support the BPS organization’s use of the BC
            Services Card integration.
          </p>
          <p class="text-left">
            We anticipate the method of compliance is through an approval and
            attestation model as follows:
          </p>
          <ol class="text-left">
            <li>
              The BPS Manager declared in the project approves the checklist
              items in a specific compliance section.
            </li>
            <li>
              Subsequently the BPS Executive Sponsor attests to each of the
              checklist items in the compliance section.
            </li>
          </ol>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component({})
export default class ProjectCompliance extends Vue {}
</script>

<style lang="scss" scoped></style>
