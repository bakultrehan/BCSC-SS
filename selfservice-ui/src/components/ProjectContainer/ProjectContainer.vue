/** * Dashboard of app */

<template>
  <v-card class="mx-auto outer-card">
    <v-container>
      <v-row class="ma-5">
        <v-col cols="12" v-if="isLoading">
          <Loading />
        </v-col>
        <v-col cols="8" v-else>
          <h2 class="text-left tab-headline">{{ projectInfo && projectInfo.projectName }}</h2>
          <h4
            class="text-left tab-headline font-weight-medium"
          >{{ projectInfo && projectInfo.organizationName }}</h4>
        </v-col>
        <v-col cols="4" class="d-flex align-end flex-column-reverse mb-2" v-show="!isLoading">
          <ProjectActions :id="id" />
        </v-col>

        <v-col cols="12">
          <v-tabs slider-color="d-none" v-model="selectedTab">
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleDevSummary')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleTeamRoles')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titlePrivacy')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleSecurity')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleComms')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleAgreements')
              }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{
              $t('projectContainer.titleLiveSummary')
              }}
            </v-tab>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <ProjectSummary :id="id" />
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <TeamRoles :id="id" />
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Live access request</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Privacy</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Security</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Access Key</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
          </v-tabs>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import ProjectSummary from '@/components/CreateApp/ProjectSummary.vue';
import Loading from '@/Atomic/Loading/Loading.vue';

import TeamRoles from '@/components/TeamRoles/TeamRoles.vue';
import ProjectActions from '@/components/ProjectActions/ProjectActions.vue';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: {
    ProjectSummary,
    Loading,
    TeamRoles,
    ProjectActions
  }
})
export default class ProjectContainer extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public projectInfo!: any;
  @ProjectInfoModule.Action('loadSingleProjectInfo')
  public loadSingleProjectInfo!: any;

  private isLoading: boolean = true;
  private selectedTab: number = 0;

  @Watch('projectInfo')
  private ongetprojectInfoChanged(val: any) {
    this.isLoading = false;
  }

  private mounted() {
    this.isLoading = true;
    if (this.projectInfo && this.projectInfo.id !== this.id) {
      this.loadSingleProjectInfo(this.id);
    }
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.custom-tabs-items {
  border: 1px solid $gray5;
  min-height: 300px; // temp fix
}
.v-tab {
  color: $BCgovFontColorInvertedDark !important;
}
.v-tab--active {
  border: 1px solid $gray5;
  background-color: $BCgovGold5;
  color: $BCgovWhite !important;
}
.tab-headline {
  color: $BCgovFontColorInvertedDark;
}
</style>
