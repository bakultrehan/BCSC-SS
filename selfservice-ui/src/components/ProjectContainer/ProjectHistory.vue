/** * ProjectHistory of app */

<template>
  <v-card class="mx-auto">
    <v-container>
      <v-row class="ma-2" v-if="isLoading">
        <Loading />
      </v-row>
      <v-row class="ma-2" v-else>
        <v-col cols="12">
          <v-col cols="12" md="12" v-if="projectHistoryData.length > 0">
            <h2 class="bc-container-title-h1 px-2 mb-5">
              {{ $t('projectHistory.tabSubTitle') }}
            </h2>

            <v-simple-table class="text-left">
              <template v-slot:default>
                <thead class="table-head">
                  <tr>
                    <th :scope="$t('projectHistory.tblTitleProjectStatus')">
                      {{ $t('projectHistory.tblTitleProjectStatus') }}
                    </th>
                    <th :scope="$t('projectHistory.tblTitleDateAssigned')">
                      {{ $t('projectHistory.tblTitleDateAssigned') }}
                    </th>
                    <th :scope="$t('projectHistory.tblActionTakenBy')">
                      {{ $t('projectHistory.tblActionTakenBy') }}
                    </th>
                    <th :scope="$t('projectHistory.tblActionRole')">
                      {{ $t('projectHistory.tblActionRole') }}
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    v-for="history in projectHistoryData"
                    :key="history.id"
                    @click="redirectToProject(project)"
                    style="cursor: pointer"
                  >
                    <td>
                      {{
                        $t(
                          `projectStatus.status${
                            projectStatusList[history.status]
                          }`
                        )
                      }}
                    </td>
                    <td>{{ history.created }}</td>
                    <td>{{ history.name }}</td>
                    <td>
                      {{
                        $t(
                          history.role &&
                            `projectRoles.role${projectRolesList[history.role]}`
                        )
                      }}
                    </td>
                  </tr>
                </tbody>
              </template>
            </v-simple-table>
          </v-col>
          <v-col cols="12" md="12" v-else>
            <v-card flat class="text-left">
              <v-card-text>{{ $t('projectHistory.noData') }}</v-card-text>
            </v-card>
          </v-col>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { projectStatus, projectRoles } from '@/constants/enums';
import VirtualCardCount from '@/components/Dashboard/VirtualCardCount.vue';
import Loading from '@/Atomic/Loading/Loading.vue';

import Button from '@/Atomic/Button/Button.vue';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: {
    Button,
    VirtualCardCount,
    Loading,
  },
})
export default class ProjectHistory extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @ProjectInfoModule.Getter('getProjectHistory')
  public getProjectHistory!: any;
  @ProjectInfoModule.Action('loadProjectHistory')
  public loadProjectHistory!: any;

  private projectStatusList: any = projectStatus;
  private projectRolesList: any = projectRoles;
  private isLoading: boolean = true;
  private projectHistoryData: any = [];

  @Watch('getProjectHistory')
  private ongetProjectHistoryChanged(val: any) {
    this.isLoading = false;
    this.projectHistoryData = val;
  }

  private mounted() {
    this.loadProjectHistory(this.id);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.table-head {
  background-color: $BCgovBlue5 !important;
  color: $BCgovWhite !important;
  & tr {
    & th {
      color: $BCgovWhite !important;
      font-size: 14px;
    }
  }
}
</style>
