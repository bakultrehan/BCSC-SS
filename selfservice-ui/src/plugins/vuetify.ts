import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
