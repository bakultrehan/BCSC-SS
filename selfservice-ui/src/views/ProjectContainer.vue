<template>
  <div class="home">
    <ProjectContainer :tab="tab" :id="id" />
  </div>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Vue, Prop } from 'vue-property-decorator';
import ProjectContainer from '@/components/ProjectContainer/ProjectContainer.vue';

@Component({
  components: {
    ProjectContainer,
  },
})
export default class ProjectContainerHome extends Vue {
  @Prop({ default: 'info' })
  public tab!: string;
  @Prop({ default: 0 })
  public id!: string;
}
</script>
