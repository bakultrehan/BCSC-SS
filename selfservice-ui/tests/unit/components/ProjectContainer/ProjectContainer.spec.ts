import { shallowMount, createLocalVue } from '@vue/test-utils';
import ProjectContainer from '@/components/ProjectContainer/ProjectContainer.vue';
import vuetify from 'vuetify';

describe('ProjectContainer.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(ProjectContainer, {
      localVue,
      mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
