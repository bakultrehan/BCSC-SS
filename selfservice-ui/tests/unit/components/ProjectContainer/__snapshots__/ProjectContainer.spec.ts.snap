// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto outer-card"
  loaderheight="4"
  tag="div"
>
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      class="ma-5"
      tag="div"
    >
      <v-col-stub
        cols="12"
        tag="div"
      >
        <loading-stub />
      </v-col-stub>
       
      <!---->
       
      <v-col-stub
        cols="12"
        tag="div"
      >
        <v-tabs-stub
          activeclass=""
          mobilebreakpoint="1264"
          nexticon="$next"
          previcon="$prev"
          slidercolor="d-none"
          slidersize="2"
          value="0"
        >
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-stub
            class="font-weight-bold"
            ripple="true"
          >
            
            
          
          </v-tab-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <projectsummary-stub
                  id="0"
                />
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <teamroles-stub
                  id="0"
                />
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <p>
                  Live access request
                </p>
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <p>
                  Privacy
                </p>
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <p>
                  Security
                </p>
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
           
          <v-tab-item-stub
            class="custom-tabs-items"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-card-text-stub>
                <p>
                  Access Key
                </p>
              </v-card-text-stub>
            </v-card-stub>
          </v-tab-item-stub>
        </v-tabs-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-card-stub>
`;
